otpInput/otpInput.js

```diff
@@ -0,0 +1,148 @@
+export class OtpInput {
+  form = document.createElement('form');
+  inputQty = 6;
+  inputs = [];
+  input = document.createElement('input');
+  otp = '';
+  inputMatchesOtp = false;
+
+  constructor(inputQty = 6) {
+    this.form.className = 'otp-form';
+    this.inputQty = inputQty;
+    this.input.type = 'hidden';
+    this.input.name = 'otp';
+  }
+
+  getOtp() {
+    return this.input.value;
+  }
+
+  setOtp(otp) {
+    this.otp = otp;
+  }
+
+  resetOtp() {
+    this.input.value = '';
+    this.otp = '';
+    this.inputs.forEach((input) => {
+      input.value = '';
+    });
+  }
+
+  createElements() {
+    for (let i = 0; i < this.inputQty; i++) {
+      const inputElem = document.createElement('input');
+      inputElem.type = 'number';
+      inputElem.className = 'otp-input';
+      inputElem.maxLength = 1;
+      inputElem.pattern = 'd*';
+      inputElem.id = `otp-input-${i}`;
+      this.form.appendChild(inputElem);
+      this.inputs.push(inputElem);
+    }
+    document.body.appendChild(this.form);
+    this.addEventListeners();
```

Comment:
Seems like you're calling `addEventListeners` twice, here and on line 145. If the expectation is that people will be calling `renderAndUse` and never calling `createElements` directly, then you shouldn't need this line.